s2stac/src/main/scala/crawler/Crawler.scala

```diff
@@ -0,0 +1,484 @@
+          yearHistory
+        )
+      }
+    }
+
+  private def makeMonthParent(history: CrawlerState, monthHistory: History.MonthHistory) =
+    if (history.months.contains(monthHistory)) {
+      printInfo[F](s"month ${monthHistory.toID} already exists") *>
+        Sync[F].pure(CrawlerState.initial)
+    } else {
+      val monthCatalog = StacCatalog(
+        "1.0.0-beta1",
+        Nil,
+        s"catalog-month-${monthHistory.toID}",
+        None,
+        s"Catalog holding descendants of month ${monthHistory.month}",
+        List(
+          StacLink("../catalog.json", StacLinkType.Parent, Some(`application/json`), None),
+          StacLink("../../../../../collection.json", StacLinkType.StacRoot, Some(`application/json`), None)
+        )
+      )
+      printInfo[F](s"month ${monthHistory.toID} did not exist!") map { _ =>
+        CrawlerState.monthHistory(monthHistory, monthCatalog) `combine` CrawlerState.yearCatalogChild(
+          YearCatalogId(monthHistory.year.toID),
+          monthHistory
+        )
+      }
+    }
+
+  private def makeDayParent(history: CrawlerState, dayHistory: History.DayHistory) =
+    if (history.days.contains(dayHistory)) {
+      printInfo[F](s"day ${dayHistory.toID} already exists") *>
+        Sync[F].pure(CrawlerState.initial)
+    } else {
+      val dayCatalog = StacCatalog(
+        "1.0.0-beta1",
+        Nil,
+        s"catalog-day-${dayHistory.toID}",
+        None,
+        s"Catalog holding descendants of day ${dayHistory.day}",
+        List(
+          StacLink("../catalog.json", StacLinkType.Parent, Some(`application/json`), None),
+          StacLink("../../../../../../collection.json", StacLinkType.StacRoot, Some(`application/json`), None)
+        )
+      )
+      printInfo[F](s"day ${dayHistory.toID} did not exist!") map { _ =>
+        CrawlerState.dayHistory(dayHistory, dayCatalog) `combine` CrawlerState.monthCatalogChild(
+          MonthCatalogId(dayHistory.month.toID),
+          dayHistory
+        )
+      }
+    }
+
+  private def getItemAssets(tileInfo: L1CTileInfo): Map[String, StacItemAsset] = {
+    Map(
+      s2Bands
+        .map({ (band: Band) =>
+          band.name.value -> StacItemAsset(
+            s"s3://${l1cBucket}/${tileInfo.path}/${band.name}.jp2",
+            band.commonName map { _.value },
+            None,
+            Set(StacAssetRole.Data),
+            Some(`image/jp2`)
+          ).addExtensionFields(
+            EOAssetExtension(NonEmptyList.one(band))
+          )
+        })
+        .toList: _*
+    )
+  }
+
+  private def makeItem(productInfo: L1CProductInfo, tileInfo: L1CTileInfo): StacItem = {
+    val reprojectedDataGeom = tileInfo.tileDataGeometry.geom.reproject(tileInfo.tileDataGeometry.crs, LatLng)
+    val dataExtent          = reprojectedDataGeom.extent
+    val eoExt               = EOItemExtension(s2Bands, Some(tileInfo.cloudyPixelPercentage.value))
+    StacItem(
+      s"${productInfo.id.value}",
+      "1.0.0-beta1",
+      Nil,
```

Comment:
I think this is extensions -- should at least be "eo", it also looks like the sentinel stac includes some properties from the "sentinel" extension, which you included here. I think that's the right move so we probably shouldn't include it in the extensions.

Reply:
woops sure is

also -- so we probably ~shouldn't~ _should_ include it? I left off `sentinel` because I can't find any evidence that a thing called a "sentinel" extension even exists. I included those fields and one extra `sentinel:` field because of what's in the Sentinel 0.6 STAC that I've been linking to

Reply:
I think leaving it off right now is the right move - we don't have an external spec to reference.

I think including the properties makes sense though.

Reply:
EO included in b85e743